ext/date/lib/dow.c

```diff
@@ -74,6 +74,36 @@ timelib_sll timelib_iso_day_of_week(timelib_sll y, timelib_sll m, timelib_sll d)
 	return timelib_day_of_week_ex(y, m, d, 1);
 }
 
+/* Checks if the given date is a weekend (Saturday or Sunday) */
+int timelib_is_weekend(timelib_time *time)
```

Comment:
nit: what is under lib should be [go upstream first](https://github.com/derickr/timelib).